subsys/net/ip/ipv4.c

```diff
@@ -91,8 +91,10 @@ struct net_pkt *net_ipv4_create(struct net_context *context,
 				   net_context_get_ip_proto(context));
 }
 
-int net_ipv4_finalize_raw(struct net_pkt *pkt, u8_t next_header)
+int net_ipv4_finalize(struct net_context *context, struct net_pkt *pkt)
```

Comment:
nack on that. 

Actually, I have been reworking this because of how net_pkt is going to be allocated. Let me send the patches (I have one for ipv6)